pytorch_lightning/trainer/trainer.py

```diff
@@ -613,6 +613,16 @@ def add_argparse_args(cls, parent_parser: ArgumentParser) -> ArgumentParser:
                     def allowed_type(x):
                         return bool(parsing.strtobool(x))
 
+                    # Bool args with default of True parsed as flags not key value pair
```

Comment:
But don't we add the argument twice to the parser in such a case? One time here, and another one there:
https://github.com/PyTorchLightning/pytorch-lightning/blob/8614074dc3d97ec5b9b3296b297010f60e36e2cb/pytorch_lightning/trainer/trainer.py#L630-L635

I think it could be handled like so:

(the code is non-working sketch)
```
if arg_types == (bool,) and arg_default is False:
    action = 'store_true'
    default = None
    arg_type = None
else:
   action = None
   default = arg_default
   arg_type = allowed_type
...
parser.add_argument(
    f'--{arg}',
    default=default,
    type=arg_type,
    dest=arg,
    action=action,
    type = type
    help='autogenerated by pl.Trainer'
)
...
```
So, I think, it could be generalized a little

Reply:
Nice, this is the way I did it originally but ended up just goin with the one I pushed. I'll change it back to something like this. Thank you!

Reply:
i removed ```parsing.strtobool(x)``` on purpose... this fails on some python versions.